File: easy/python/0111.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def minDepth(self, root: 'TreeNode') -> 'int':
        if not root:
            return 0
        queue = [(root, 1)]
        while queue:
            next_, level = queue[0]
            queue = queue[1:]
            left, right = next_.left, next_.right
            if not left and not right:
                return level
            if left:
                queue.append((left, level+1))
            if right:
                queue.append((right, level+1))

File: easy/cpp/0111.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int minDepth(TreeNode* root) {
        if(root == NULL) return 0;
        deque<TreeNode*> queue;
        root->val = 1;
        queue.push_back(root);
        while(queue.size() != 0){
            TreeNode* next = queue[0];
            queue.pop_front();
            if(next->left == NULL && next->right == NULL) return next->val;
            if(next->left != NULL){
                next->left->val = next->val + 1;
                queue.push_back(next->left);
            }
            if(next->right != NULL){
                next->right->val = next->val + 1;
                queue.push_back(next->right);
            }    
        }
        return 0;
    }
};